internal/adapters/consul/sync_test.go

```diff
@@ -202,6 +202,7 @@ func TestConsulSyncAdapter_Sync(t *testing.T) {
 		require.True(t, ok)
 		require.NotNil(t, ingress)
 
-		return ingress.Listeners[0].TLS.TLSMinVersion == "TLSv1_2"
+		return ingress.Listeners[0].TLS.TLSMinVersion == "TLSv1_2" &&
+			reflect.DeepEqual(ingress.Listeners[0].TLS.CipherSuites, common.DefaultTLSCipherSuites())
```

Comment:
do we know if Consul maintains the ordering here? `reflect.DeepEqual` returns `false` if the order of an array changes IIRC

Reply:
The sync adapter assigns these directly to the config entry at https://github.com/hashicorp/consul-api-gateway/pull/71/files#diff-9defe7ecca2a3bf51ee560c795c29ba0d8c95dfef3ef8c864aeab9e85a5ec97fR199 and AFAIK Consul should preserve the ordering of values in an array (as opposed to something like map key iteration order which won't be guaranteed).

Envoy appears to rely on [BoringSSL's cipher suite negotiation](https://boringssl.googlesource.com/boringssl/+/refs/heads/master/ssl/ssl_cipher.cc) which ensures predefined ordering to prefer more secure cipher suites so preserving the ordering here isn't _essential_ as it may be for other proxy config, but I don't believe it should change.